[RPC Framework] Support retrieving the RRef to the remote module

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48983 [RPC Framework] Support retrieving the RRef to the remote module**

Expose an API for users to retrieve the RRef for the underlying module.

This would be useful if users would like to run custom code on the remote end for the nn.Module.

Original PR issue: RemoteModule enhancements #40550

Differential Revision: [D25386042](https://our.internmc.facebook.com/intern/diff/D25386042/)

Commit:
[RPC Framework] Support retrieving the RRef of the remote module

Expose an API for users to retrieve the RRef for the underlying module.

This would be useful if users would like to run custom code on the remote end for the nn.Module.

Original PR issue: RemoteModule enhancements #40550

Differential Revision: [D25386042](https://our.internmc.facebook.com/intern/diff/D25386042/)

[ghstack-poisoned]

pritamdamania87 commented on torch/distributed/nn/api/remote_module.py:
```diff
@@ -209,6 +209,9 @@ def remote_parameters(self, recurse: bool = True) -> List[rpc.RRef[Parameter]]:
         """
         return rpc.rpc_sync(self.on, _param_rrefs, args=(self.module_rref, recurse))
 
+    def get_module_rref(self) -> rpc.RRef[nn.Module]:
+        return self.module_rref
+
```
Add docs for this method

Commit:
Update on "[RPC Framework] Support retrieving the RRef to the remote module"


Expose an API for users to retrieve the RRef for the underlying module.

This would be useful if users would like to run custom code on the remote end for the nn.Module.

Original PR issue: RemoteModule enhancements #40550

Differential Revision: [D25386042](https://our.internmc.facebook.com/intern/diff/D25386042/)

[ghstack-poisoned]
